overlord/hookstate/hooks.go

```diff
@@ -234,25 +236,27 @@ func (h *gateAutoRefreshHookHandler) Error(hookErr error) (err error) {
 
 	var affecting []string
 	if err := ctx.Get("affecting-snaps", &affecting); err != nil {
-		return fmt.Errorf("internal error: cannot get affecting-snaps")
+		return false, fmt.Errorf("internal error: cannot get affecting-snaps")
 	}
 
 	// no duration specified, use maximum allowed for this gating snap.
 	var holdDuration time.Duration
 	if err := snapstate.HoldRefresh(st, snapName, holdDuration, affecting...); err != nil {
-		// note, previous hook error (hookErr) is going to be logged by hookmgr
-		// after this Error() handler, so only log the new error.
-		h.context.Errorf("error: %v (while handling previous hook error)", err)
+		// log the original hook error as we either ignore it or error out from
+		// this handler, in both cases hookErr won't be logged by hook manager.
+		h.context.Errorf("error: %v (while handling previous hook error: %v)", err, hookErr)
 		if _, ok := err.(*snapstate.HoldError); ok {
-			return nil
+			return true, nil
```

Comment:
we probably want to leave a TODO here that like for the snap themselves we want to be able to delay for another hour maybe in this case?